### PR TITLE
Don't break on Cargo.toml without package.metadata.sphinx. (Fixes #26)

### LINKED ISSUE #26: cargo-sphinx fails on projects without Cargo.toml configuration set.
```
$ cargo new fatal
$ cd fatal
$ cargo sphinx
error: Fatal: Invalid cargo file format.
```

### PR DESCRIPTION
**Pull request by daithiocrualaoich**

### CODE REVIEW

**Comment by daithiocrualaoich** [comment]
Too bad, borrowchecker. Better luck next time!

File: src/config.rs [hunk 1]
```diff
@@ -66,18 +66,20 @@ impl Config {
...
 
         let mut parser = Parser::new(&contents);
+        let parsed: Option<Table> = parser.parse();
 
-        let toml: Option<Table> = parser.parse();
-        let sphinx_toml = toml.and_then(|mut table| table.remove("package"))
+        // Verify parsed TOML is valid.
+        let mut toml: Table = try!(parsed.ok_or(FatalError::InvalidCargoFileFormat));
+
+        let config: Table = toml.remove("package")
```
Comment (passy) [inline src/config.rs]: Ooh!

**Comment by passy** [comment]
Well done. Ship that!

File: src/config.rs (hunk 1 above)
Comment (daithiocrualaoich) [inline src/config.rs]: The truly dark and sneaky bit is below in the `unwrap_or(Table::new())` instead of fighting for `Option<Table>` in the `struct`. :)

**Pull request merged**